unitTests/ui/ResultListTest.ts

```diff
@@ -784,11 +784,58 @@ export function ResultListTest() {
         describe('enableScrollToTop set to', () => {
           let infiniteScrollContainer: HTMLElement;
           let allResult: IQueryResults;
+          let scrollToCalled: boolean;
+
           beforeEach(() => {
-            infiniteScrollContainer = { scrollTop: 500 } as HTMLElement;
+            scrollToCalled = false;
+            infiniteScrollContainer = ({
+              scrollTop: 500,
+              scrollTo: (x: number, y: number) => {
+                scrollToCalled = true;
+                infiniteScrollContainer.scrollTop = x;
+              }
+            } as any) as HTMLElement;
```

Comment:
Instead of casting to `any` to bypass Typescript, would there be any harm in creating a real HTMLElement?

Also, rather than using a property `scrollToCalled`, could we spy on the scrollTo method, or override the method with a spy?

Related, do we need to customise the implementation? `infiniteScrollContainer.scrollTop = x;`?
If it's just to get the arg passed to `scrollTo`, I think we can use the spy `toHaveBeenCalledWith` method.

I think the most important things to check are that 1) `scrollTo` is called, and 2) it is called with the right value?

Reply:
👍